File: 프로그래머스/lv2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        Arrays.sort(citations);
        int answer = 0;
        for (int i = citations.length; i >= 0; i--) {
            int biggerCount = 0, lowerCount = 0;
            for (int j = 0; j < citations.length; j++) {
                if (i >= citations[j]) {
                    lowerCount++;
                }
                if (i <= citations[j]) {
                    biggerCount++;
                }   
            }
            if (i >= lowerCount && i <= biggerCount) {
                answer = i;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<int> citations) {
    
    sort(citations.begin(), citations.end(), greater<int>());
    
    if (citations[0] == 0) {
        return 0;
    }
    
    int answer;
    for (int i = 0; i < citations.size(); i++) {
        
        if (citations[i] >= i+1) {
            answer =  i+1;
        }
    }
    
    return answer;
}